Libs/MRML/Core/vtkMRMLSubjectHierarchyNode.cxx

```diff
@@ -1119,6 +1119,9 @@ bool vtkSubjectHierarchyItem::RemoveChild(vtkSubjectHierarchyItem* item)
   // as they represent the item's data node's content), then remove virtual branch
   if (removedItem->IsVirtualBranchParent())
     {
+    // If the item has many virtual children, then it may invoke a large number of events.
+    // To avoid this, block modified events on the data node until all of the children are removed.
+    MRMLNodeModifyBlocker blocker(removedItem->DataNode);
     removedItem->RemoveAllChildren();
```

Comment:
In addition of `tkSubjectHierarchyItem::RemoveChild(vtkSubjectHierarchyItem* item)`, should this be done in these functions as well where `RemoveAllChildren` is also called? 
* `vtkSubjectHierarchyItem::DeepCopy(vtkSubjectHierarchyItem* item, bool copyChildren/*=true*/)`
* `vtkSubjectHierarchyItem::~vtkSubjectHierarchyItem()`
* `vtkSubjectHierarchyItem::RemoveChild(vtkIdType itemID)`
* `vtkMRMLSubjectHierarchyNode::Copy(vtkMRMLNode *anode)`

Even better, should we consider updating `RemoveAllChildren` to systematically block the event ?

Reply:
That's a good point, the same performance issue could occur with any of those functions.
I've moved the MRMLNodeModifyBlocker into RemoveAllChildren.